File: Java/Bink/src/net/qiujuer/blink/core/Receiver.java
package net.qiujuer.blink.core;

/**
 * Receive entity interface
 */
public interface Receiver {

    /**
     * Receive the entity's information
     *
     * @return ReceiveEntity
     */
    ReceivePacket<?> receiveHead();


    /**
     * Receive entity
     *
     * @param entity   ReceiveEntity
     * @param delivery ReceiveDelivery
     * @return ReceiveEntity
     */
    boolean receiveEntity(ReceivePacket<?> entity, ReceiveDelivery delivery);

    /**
     * Destroy the receive IO stream
     */
    void destroyReceiveIO();
}

File: C Sharp/Blink/Blink/Core/Receiver.cs
using System;
using System.Net.Sockets;
namespace Net.Qiujuer.Blink.Core
{
    /**
     * Receive entity interface
     */
    public interface Receiver : IDisposable
    {
        /// <summary>
        /// Get receive buffer size
        /// </summary>
        /// <returns>Buffer Size</returns>
        int GetBufferSize();

        /// <summary>
        /// Async receive some data to buffer
        /// </summary>
        /// <param name="e">SocketAsyncEventArgs</param>
        /// <returns>Status</returns>
        bool ReceiveAsync(SocketAsyncEventArgs e);
    }
}
